docs/src/tutorials/transitioning/transitioning_from_matlab.jl

```diff
@@ -0,0 +1,447 @@
+# of the matrix:
+
+x_sym = LinearAlgebra.Symmetric(x, :L)
+
+#-
+
+x_sym = LinearAlgebra.Symmetric(x, :U)
+
+# The same applies for Hermitian matrices, using `LinearAlgebra.Hermitian` and
+# `LinearAlgebra.ishermitian`.
+
+# ## Primal versus dual form
+
+# When you translate some optimization problems from YALMIP or CVX to JuMP, you
+# might be surprised to see it get much faster or much slower, even if you're
+# using exactly the same solver. The most likely reason is that YALMIP will
+# always interpret the problem as the dual form, whereas CVX and JuMP will try to
+# interpret the problem in the form most appropriate to the solver. If the
+# problem is more naturally formulated in the primal form it is likely that
+# YALMIP's performance will suffer, or if JuMP gets it wrong, its performance will
+# suffer. It might be worth trying both primal and dual forms if you're having
+# trouble, which can be done automatically with the package [Dualization.jl](@ref).
+#
+# For an in-depth explanation of this issue, see the [Dualization](@ref) tutorial.
+# ## Rosetta stone
+
+# In this section, we show a complete example of the same optimization problem
+# being solved with JuMP, YALMIP, and CVX. It is a semidefinite program that
+# computes a lower bound on the random robustness of entanglement using the
+# partial transposition criterion.
+
+# The code is complete, apart from the function that does partial transposition.
+# With both YALMIP and CVX we use the function `PartialTranspose` from
+# [QETLAB](https://github.com/nathanieljohnston/QETLAB). With JuMP, we could use
+# the function `Convex.partialtranspose` from [Convex.jl](https://jump.dev/Convex.jl/stable/),
+# but we reproduce it here for simplicity:
+
+function partial_transpose(x::AbstractMatrix, sys::Int, dims::Vector)
+    @assert size(x, 1) == size(x, 2) == prod(dims)
+    @assert 1 <= sys <= length(dims)
+    n = length(dims)
+    s = n - sys + 1
+    p = collect(1:2n)
+    p[s], p[n+s] = n + s, s
+    r = reshape(x, (reverse(dims)..., reverse(dims)...))
+    return reshape(permutedims(r, p), size(x))
+end
+
+# ### JuMP
+
+# The JuMP code to solve this problem is:
+
+using JuMP
+import Clarabel
+import LinearAlgebra
+
+function random_state_pure(d)
+    x = randn(Complex{Float64}, d)
+    y = x * x'
+    return LinearAlgebra.Hermitian(y / LinearAlgebra.tr(y))
+end
+
+function robustness_jump(d)
+    rho = random_state_pure(d^2)
+    id = LinearAlgebra.Hermitian(LinearAlgebra.I(d^2))
+    rhoT = LinearAlgebra.Hermitian(partial_transpose(rho, 1, [d, d]))
+    model = Model()
+    @variable(model, λ)
+    @constraint(model, PPT, rhoT + λ * id in HermitianPSDCone())
+    @objective(model, Min, λ)
+    set_optimizer(model, Clarabel.Optimizer)
+    set_attribute(model, "verbose", true)
+    optimize!(model)
+    if is_solved_and_feasible(model)
+        WT = dual(PPT)
+        return value(λ), real(LinearAlgebra.dot(WT, rhoT))
```

Comment:
Why did add `real` here? It's redundant, `dot` already returns a real number from Julia 1.11 onwards.

Reply:
Because the docs currently build with Julia 1.10 and we support Julia 1.6. Also, v1.11 isn't released yet.

Reply:
What's the problem? It doesn't cause any error with older versions of Julia, the output is just slightly less neat.

Reply:
`dot` currently returns a `Complex{Float64}` instead of a `Float64`. Your MATLAB examples return a `Float64`.

```julia
julia> import LinearAlgebra

julia> A = LinearAlgebra.Hermitian(rand(ComplexF64, (3, 3)))
3×3 LinearAlgebra.Hermitian{ComplexF64, Matrix{ComplexF64}}:
 0.468242+0.0im       0.296799+0.49972im   0.621408+0.148878im
 0.296799-0.49972im   0.413853+0.0im       0.677598+0.875476im
 0.621408-0.148878im  0.677598-0.875476im   0.59341+0.0im

julia> LinearAlgebra.dot(A, A)
4.6860981128420285 + 0.0im

julia> real(LinearAlgebra.dot(A, A))
4.6860981128420285
```

Reply:
I know. This is not an error, and will change anyway in Julia 1.11.

Reply:
> I know

I know because I saw your PR :smile:

My point is that I personally would argue that changing from `ComplexF64` to `Float64` is a breaking change, and the fact it will change in Julia v1.11 is not compelling because we still have many JuMP users on a range of old versions. I want something that works across all versions, not just some upcoming one. We cannot assume that new users from MATLAB will use a recent version of Julia. See, for example, a [discourse user on Julia v1.5](https://discourse.julialang.org/t/fail-to-update-gurobi-jl-for-using-gurobi-11-0/111225).

Reply:
I really don't understand the problem. It works. It's just slightly uglier. And moreover it's just example code in a tutorial. Nothing depends on whether its return value is `Float64` or `ComplexF64`.

Reply:
It doesn't work. One return a `ComplexF64`, and the other returns a `Float64`.

Reply:
Why is this a problem?